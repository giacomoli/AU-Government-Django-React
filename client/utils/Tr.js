import React from 'react'
import PropTypes from 'prop-types'

const Tr = (props) => {
  const _onClick = () => props.onClick(props.value)

  return (
    <tr {...props} onClick={_onClick}>
      {props.children}
      <style jsx>{`
        tr {
          cursor: pointer;
        }
      `}</style>
    </tr>
  )
}

Tr.propTypes = {
  onClick: PropTypes.func.isRequired,
}

export default Tr
