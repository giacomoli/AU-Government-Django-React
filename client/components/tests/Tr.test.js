import React from 'react'
import renderer from 'react-test-renderer'
import Tr from '../../utils/Tr'

/* eslint-disable */
describe('Tr', () => {

  it('should render Tr with props', () => {
    const props = {
      onClick: jest.fn(),
      children: (<th>test</th>)
    }
    const wrapper = renderer.create(
        <Tr
          props={props}
          onClick={jest.fn()}
        />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render Tr without props children', () => {
    const props = {
      onClick: jest.fn(),
      children: (<th>test</th>)
    }
    const wrapper = renderer.create(
        <Tr
          onClick={jest.fn()}
        />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
