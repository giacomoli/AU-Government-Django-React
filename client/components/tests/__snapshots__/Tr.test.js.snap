// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tr should render Tr with props 1`] = `
<tr
  className="jsx-391693574 "
  onClick={[Function]}
  props={
    Object {
      "children": <th>
        test
      </th>,
      "onClick": [MockFunction],
    }
  }
/>
`;

exports[`Tr should render Tr without props children 1`] = `
<tr
  className="jsx-391693574 "
  onClick={[Function]}
/>
`;
